0024-pre-shared-key-v2/0024-pre-shared-key-v2.md

```diff
@@ -0,0 +1,266 @@
+
+If senders want to use a path with an MPS lower than their payment amount, they need to split their payment into chunks. PSKv2 includes a Payment ID and Sequence number to enable receivers to associate incoming payment chunks properly.
+
+**Possibility for Incomplete Payments:** If the path between the sender and receiver runs out of liquidity while a chunked payment is in progress, it could be impossible for the payment to complete fully. This should be a rare case, analogous to Internet Service Providers getting cut off because they fail to pay their electricity bills. However, in such cases, the receiver may need to refund the partial payment by sending another payment back and either the sender or receiver would incur the transaction costs.
+
+**Fluctuating Exchange Rates:** The rate from the sender to the receiver may change due to legitimate market movements during the course of a payment. Senders SHOULD monitor the rate changes and they may not want to send payments that are significantly larger than their bandwidth (this could be compared to how downloading a 4K movie over a dial-up connection is possible but would not make for a good experience).
+
+**Malicious Connectors:** Exchange rates can also change for illegitimate reasons, such as connectors driving up the price mid-payment. Ultimately, the main force keeping connectors' rates low is competition. If there is only one path, the connectors do not need to do anything sneaky to drive up the rate. If a sender has multiple options for the connector they send through, they may want to use [Multi-Path Chunked Payments](#multi-path-chunked-payments) to automatically switch paths if one becomes more expensive.
+
+**Trusting the Receiver:** When doing any Interledger payment aside from an atomic exchange of money for goods or services, the sender must trust the receiver to deliver the goods once they receive the payment. With chunked payments, the sender must trust the receiver to deliver the goods and accurately report how much money has arrived on their incoming transfer. A malicious receiver could keep reporting less money having arrived on every transaction in an attempt to extract more money from the sender, but this is no worse a threat than a receiver demanding more money after a non-chunked payment arrives but before they release the goods or services. Since senders will judge merchants' prices by the total source amount they pay, receivers that pull such tricks should develop negative reputations and be avoided.
+
+## Model of Operation
+
+### Informational Quote
+
+1. Sender creates a packet with a type of 1, a random Payment ID, a Sequence number of 0, a Payment Amount of 0, a Chunk Amount set to 18446744073709551615 (the maximum value) and [encrypts it](#encryption-envelope).
```

Comment:
By payment amount and chunk amount, do you mean transfer amount and packet amount? Why would anyone forward such a payment if the transfer amount is zero?

Reply:
No, it means the fields called Payment Amount and Chunk Amount in the PSK data packet defined below